blackbox/docs/sql/administration/user_defined_functions.txt

```diff
@@ -0,0 +1,369 @@
+    ...        return Math.sqrt( Math.pow(end_point[0] - start_point[0], 2), Math.pow(end_point[1] - start_point[1], 2));
+    ...      }';
+    CREATE OK, 1 row affected  (... sec)
+
+
+.. note::
+
+    Argument names are used for query documentation purposes only. You cannot
+    reference arguments by name in the function body.
+
+Optionally, you can specify a schema for the function. If you omit the schema,
+the current session schema is used.
+
+You can explicitly assign a schema like this::
+
+    cr> CREATE OR REPLACE FUNCTION my_schema.log10(long)
+    ...  RETURNS double
+    ...  LANGUAGE JAVASCRIPT
+    ...  AS 'function log10(a) { return Math.log(a)/Math.log(10); }';
+    CREATE OK, 1 row affected  (... sec)
+
+
+.. warning::
+
+    :ref:`snapshot-restore` can't be used to backup functions, because
+    snapshots contain table data only.
+
+Supported Types
+---------------
+
+The argument types, and the return type of the function can be any of the
+CrateDB supported :ref:`sql_ddl_datatypes`. Data types of values passed into a
+function must strictly correspond to its argument data types.
+
+.. note::
+
+    The value returned by the function will be casted to the return type
+    provided in the definition if required. An exception will be thrown
+    if the cast is not successful.
+
+
+OVERLOADING
+-----------
+
+Within a specific schema, you can overload functions by defining two functions with the
+same name that have a different set of arguments::
+
+    cr> CREATE FUNCTION my_schema.multiply(integer, integer)
+    ...  RETURNS integer
+    ...  LANGUAGE JAVASCRIPT
+    ...  AS 'function multiply(a, b) { return a * b; }';
+    CREATE OK, 1 row affected  (... sec)
+
+This would overload our ``multiply`` function with different argument types::
+
+    cr> CREATE FUNCTION my_schema.multiply(long, long)
+    ...  RETURNS long
+    ...  LANGUAGE JAVASCRIPT
+    ...  AS 'function multiply(a, b) { return a * b; }';
+    CREATE OK, 1 row affected  (... sec)
+
+This would overload our ``multiply`` function with more arguments::
+
+    cr> CREATE FUNCTION my_schema.multiply(long, long, long)
+    ...  RETURNS long
+    ...  LANGUAGE JAVASCRIPT
+    ...  AS 'function multiply(a, b, c) { return a * b * c; }';
+    CREATE OK, 1 row affected  (... sec)
+
+.. note::
+
+    It is considered bad practice to create functions that have the same name
+    as the CrateDB built-in functions. 
+    
+    If you call a function without a schema
+    name, CrateDB will look it up in the built-in functions first and only then
+    in the user-defined functions with the schema of the current session (see
+    :ref:`conf-session-settings`). 
+    
+    Therefore a built-in function with the same
+    name as a user-defined function will hide the latter. However, such
+    functions can still be called if the schema name is explicitly provided.
+
+
+Determinism
+-----------
+
+User-defined functions are deterministic, meaning that they will always return
```

Comment:
if this is the case, then you should write that the functions are required to be deterministic, since this depends on the users implementation of the function. this should be a warning.